debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlReadOnlyIncrementalSnapshotContext.java

```diff
@@ -145,4 +153,8 @@ public ExecuteSnapshotKafkaSignal getExecuteSnapshotSignals() {
     public boolean hasExecuteSnapshotSignals() {
         return !executeSnapshotSignals.isEmpty();
     }
+
+    public boolean watermarksChanged() {
+        return !previousLowWatermark.equals(lowWatermark) || !previousHighWatermark.equals(highWatermark);
```

Comment:
Is it guaranteed that previous watermarks will be always not null?

Reply:
Yes, this watermarks change check is done after the `closeWindow()` call, so previous watermarks will always be initialized

Reply:
ok, thanks